src/compiler/parser.ts

```diff
@@ -3210,6 +3285,155 @@ module ts {
+                node.left = elementName;
+                node.right = parseIdentifierName();
+                elementName = finishNode(node);
+            }
+            return elementName;
+        }
+
+        function parseJsxExpression(): JsxExpression {
+            let node = <JsxExpression>createNode(SyntaxKind.JsxExpression);
+            let isExpressionEmpty = lookAhead(() => {
+                do {
+                    nextToken();
+                    if (token === SyntaxKind.CloseBraceToken) {
+                        return true;
+                    }
+                    else if (!isTrivia(token)) {
+                        return false;
+                    }
+                } while (token != SyntaxKind.EndOfFileToken);
+                return true;
+            });
+
+            parseExpected(SyntaxKind.OpenBraceToken);
+            if (!isExpressionEmpty) {
+                node.expression = parseExpression();
+            }
+            parseExpected(SyntaxKind.CloseBraceToken);
+
+            return finishNode(node);
+        }
+
+        function parseJsxAttribute(): JsxAttribute | JsxSpreadAttribute {
+            if (token === SyntaxKind.OpenBraceToken) {
+                return parseJsxSpreadAttribute();
+            }
+            reScanJsxIdentifier();
+            let node = <JsxAttribute>createNode(SyntaxKind.JsxAttribute);
+            node.name = parseIdentifierName();
+            if (parseOptional(SyntaxKind.EqualsToken)) {
+                switch (token) {
+                    case SyntaxKind.LessThanToken:
+                        node.initializer = parseJsxElement();
+                        break;
+                    case SyntaxKind.StringLiteral:
+                        node.initializer = parseLiteralNode();
+                        break;
+                    default:
+                        node.initializer = parseJsxExpression();
+                        break;
+                }
+            }
+            return finishNode(node);
+        }
+
+        function parseJsxSpreadAttribute(): JsxSpreadAttribute {
+            let node = <JsxSpreadAttribute>createNode(SyntaxKind.JsxSpreadAttribute);
+            parseExpected(SyntaxKind.OpenBraceToken);
+            parseExpected(SyntaxKind.DotDotDotToken);
+            node.expression = parseExpression();
+            parseExpected(SyntaxKind.CloseBraceToken);
+            return finishNode(node);
+        }
+
+        function parseJsxClosingElement(expectedTagName: string): JsxClosingElement {
+            let node = <JsxClosingElement>createNode(SyntaxKind.JsxClosingElement);
+            parseExpected(SyntaxKind.LessThanSlashToken);
+            node.tagName = parseJsxElementName();
+            if (entityNameToString(node.tagName) !== expectedTagName) {
+                parseErrorAtCurrentToken(Diagnostics.Expected_corresponding_JSX_closing_tag_for_0, expectedTagName);

```

Comment:
Nope.  Make this a checker error.  If you do things this way you'll definitely break incremental parsing (even if the edit doesn't touch the Xml element).